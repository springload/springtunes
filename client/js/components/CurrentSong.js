import React, { PropTypes, Component } from 'react';
import Processing from '../components/Processing';
import Icon from '../components/Icon';

export default class CurrentSong extends Component {
    render() {
        const { currentSong, isFetching, error } = this.props;

        return (
            <div className='current-song'>
                <div className='song-details'>
                    <div className='song-details__graphic'>
                        <Icon name='fa-headphones' iconType='big' />
                    </div>
                    <div className='song-details__content'>
                        <div className='twelve columns'>
                            { !isFetching &&
                                <div>
                                    <h2>
                                        <a
                                            href={ currentSong.link_track }
                                            target='_blank'
                                            title={ `View track: ${currentSong.title}` }
                                        >
                                            { currentSong.title }
                                        </a>
                                    </h2>
                                    <h3>
                                        <a
                                            href={ currentSong.link_artist }
                                            target='_blank'
                                            title={ `View Artist: ${currentSong.artist}` }
                                        >
                                            { currentSong.artist }
                                        </a>
                                    </h3>
                                    <h3>
                                        <a
                                            href={ currentSong.link_album }
                                            target='_blank'
                                            className='color-text-cement'
                                            title={ `View Album: ${currentSong.album}` }
                                        >
                                            { currentSong.album }
                                        </a>
                                    </h3>
                                </div>
                            }
                            { isFetching &&
                                <Processing />
                            }
                            { !isFetching && error !== '' &&
                                <div className='error'>
                                    Error occurred.<br />
                                    Details are: "{ error }"
                                </div>
                            }
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

CurrentSong.propTypes = {
    isFetching: PropTypes.bool.isRequired,
    currentSong: PropTypes.object.isRequired,
    error: PropTypes.string,
};
