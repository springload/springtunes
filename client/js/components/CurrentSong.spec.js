import React from 'react';
import CurrentSong from './CurrentSong';
import ReactTestUtils from 'react-addons-test-utils';

describe('CurrentSong component', () => {
    it('renders correctly with min params', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        const result = shallowRenderer.render(
            <CurrentSong
                isFetching={false}
                currentSong={
                    {
                        title: "Test song",
                        artist: "Test artist",
                        album: "Test album",
                        link_track: "http://test.com",
                        link_artist: "http://test.com",
                        link_album: "http://test.com",
                    }
                }
            />
        );
        expect(result).toMatchSnapshot();
    });

    it('renders correctly when fetching', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        const result = shallowRenderer.render(
            <CurrentSong
                isFetching={true}
                currentSong={
                    {
                        title: "Test song",
                        artist: "Test artist",
                        album: "Test album",
                        link_track: "http://test.com",
                        link_artist: "http://test.com",
                        link_album: "http://test.com",
                    }
                }
            />
        );
        expect(result).toMatchSnapshot();
    });

    it('renders correctly when error', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        const result = shallowRenderer.render(
            <CurrentSong
                isFetching={false}
                currentSong={
                    {
                        title: "Test song",
                        artist: "Test artist",
                        album: "Test album",
                        link_track: "http://test.com",
                        link_artist: "http://test.com",
                        link_album: "http://test.com",
                    }
                }
                hasError={true}
                error="ERROR"
            />
        );
        expect(result).toMatchSnapshot();
    });
});
