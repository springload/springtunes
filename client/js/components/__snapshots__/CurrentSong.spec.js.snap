exports[`CurrentSong component renders correctly when error 1`] = `
<div
  className="current-song">
  <div
    className="song-details">
    <div
      className="song-details__graphic">
      <Unknown
        iconType="big"
        name="fa-headphones" />
    </div>
    <div
      className="song-details__content">
      <div
        className="twelve columns">
        <div>
          <h2>
            <a
              href="http://test.com"
              target="_blank"
              title="View track: Test song">
              Test song
            </a>
          </h2>
          <h3>
            <a
              href="http://test.com"
              target="_blank"
              title="View Artist: Test artist">
              Test artist
            </a>
          </h3>
          <h3>
            <a
              className="color-text-cement"
              href="http://test.com"
              target="_blank"
              title="View Album: Test album">
              Test album
            </a>
          </h3>
        </div>
        <div
          className="error">
          Error occurred.
          <br />
          Details are: \"
          ERROR
          \"
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CurrentSong component renders correctly when fetching 1`] = `
<div
  className="current-song">
  <div
    className="song-details">
    <div
      className="song-details__graphic">
      <Unknown
        iconType="big"
        name="fa-headphones" />
    </div>
    <div
      className="song-details__content">
      <div
        className="twelve columns">
        <Unknown />
      </div>
    </div>
  </div>
</div>
`;

exports[`CurrentSong component renders correctly with min params 1`] = `
<div
  className="current-song">
  <div
    className="song-details">
    <div
      className="song-details__graphic">
      <Unknown
        iconType="big"
        name="fa-headphones" />
    </div>
    <div
      className="song-details__content">
      <div
        className="twelve columns">
        <div>
          <h2>
            <a
              href="http://test.com"
              target="_blank"
              title="View track: Test song">
              Test song
            </a>
          </h2>
          <h3>
            <a
              href="http://test.com"
              target="_blank"
              title="View Artist: Test artist">
              Test artist
            </a>
          </h3>
          <h3>
            <a
              className="color-text-cement"
              href="http://test.com"
              target="_blank"
              title="View Album: Test album">
              Test album
            </a>
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;
